Mesh_3/examples/Mesh_3/mesh_3D_image_with_input_features.cpp

```diff
@@ -0,0 +1,82 @@
+#include <vector>
+#include <iostream>
+
+#include <CGAL/Mesh_triangulation_3.h>
+#include <CGAL/Mesh_complex_3_in_triangulation_3.h>
+#include <CGAL/Mesh_criteria_3.h>
+
+#include <CGAL/make_mesh_3.h>
+#include <CGAL/Image_3.h>
+
+/// [Domain definition]
+#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
+#include <CGAL/Mesh_domain_with_polyline_features_3.h>
+#include <CGAL/Labeled_mesh_domain_3.h>
+
+typedef CGAL::Exact_predicates_inexact_constructions_kernel K;
+typedef CGAL::Labeled_mesh_domain_3<K> Image_domain;
+typedef CGAL::Mesh_domain_with_polyline_features_3<Image_domain> Mesh_domain;
+/// [Domain definition]
+
+#ifdef CGAL_CONCURRENT_MESH_3
+using Concurrency_tag = CGAL::Parallel_tag;
+#else
+using Concurrency_tag = CGAL::Sequential_tag;
+#endif
+
+// Triangulation
+typedef CGAL::Mesh_triangulation_3<Mesh_domain,CGAL::Default,Concurrency_tag>::type Tr;
+
+typedef CGAL::Mesh_complex_3_in_triangulation_3<Tr> C3t3;
+
+// Criteria
+typedef CGAL::Mesh_criteria_3<Tr> Mesh_criteria;
+
+namespace params = CGAL::parameters;
+
+#include "read_polylines.h"
+
+int main(int argc, char* argv[])
+{
+  const std::string fname = (argc>1)?argv[1]:CGAL::data_file_path("images/420.inr");
+  // Loads image
+  CGAL::Image_3 image;
+  if(!image.read(fname)){
+    std::cerr << "Error: Cannot read file " <<  fname << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  /// Declare 1D-features
+  const std::string lines_fname = (argc>2)?argv[2]:CGAL::data_file_path("images/420.polylines.txt");
+  using Point_3 = K::Point_3;
+  std::vector<std::vector<Point_3> > features_inside;
+  if (!read_polylines(lines_fname, features_inside)) // see file "read_polylines.h"
+  {
+    std::cerr << "Error: Cannot read file " << lines_fname << std::endl;
+    return EXIT_FAILURE;
+  }
+
+  /// [Domain creation]
+  Mesh_domain domain = Mesh_domain::create_labeled_image_mesh_domain(image,
+    params::input_features = std::cref(features_inside));//use std::cref to avoid a copy
```

Comment:
@lrineau there is an extra comment here about `std::cref`
What do you think of duplicating it?

Reply:
done